deploy/helm/sumologic/upgrade-2.0.0.sh

```diff
@@ -0,0 +1,270 @@
+}
+
+function create_temp_file() {
+  echo -n > "${TEMP_FILE}"
+}
+
+function migrate_prometheus_operator_to_kube_prometheus_stack() {
+  info "Migrating prometheus-config-reloader container to config-reloader in prometheusSpec"
+  yq m -i --arrays append \
+    "${TEMP_FILE}" \
+    <(
+      yq p <(
+        yq w <(
+          yq r "${TEMP_FILE}" -- 'prometheus-operator.prometheus.prometheusSpec.containers.(name==prometheus-config-reloader)' \
+          ) name config-reloader \
+        ) 'prometheus-operator.prometheus.prometheusSpec.containers[+]'
+      )
+  yq d -i "${TEMP_FILE}" "prometheus-operator.prometheus.prometheusSpec.containers.(name==prometheus-config-reloader)"
+
+  info "Migrating from prometheus-operator to kube-prometheus-stack"
+  yq m -i \
+    "${TEMP_FILE}" \
+    <(
+      yq p \
+      <(yq r "${TEMP_FILE}" "prometheus-operator") \
+      "kube-prometheus-stack" \
+    )
+  yq d -i "${TEMP_FILE}" "prometheus-operator"
+}
+
+function delete_migrated_unused_keys() {
+  IFS=$'\n' read -r -d ' ' -a MAPPINGS_KEYS_TO_DELETE <<< "${KEYS_TO_DELETE}"
+  readonly MAPPINGS_KEYS_TO_DELETE
+  for i in "${MAPPINGS_KEYS_TO_DELETE[@]}"; do
+    yq d -i "${TEMP_FILE}" -- "${i}"
+  done
+}
+
+function migrate_customer_keys() {
+  # Convert variables to arrays
+  set +e
+  IFS=$'\n' read -r -d ' ' -a MAPPINGS <<< "${KEY_MAPPINGS}"
+  readonly MAPPINGS
+  IFS=$'\n' read -r -d ' ' -a MAPPINGS_KEY_VALUE <<< "${KEY_VALUE_MAPPINGS}"
+  readonly MAPPINGS_KEY_VALUE
+  set -e
+
+  readonly CUSTOMER_KEYS=$(yq --printMode p r "${OLD_VALUES_YAML}" -- '**')
+
+  for key in ${CUSTOMER_KEYS}; do
+    if [[ ${MAPPINGS[*]} =~ ${key}: ]]; then
+      # whatever you want to do when arr contains value
+      for i in "${MAPPINGS[@]}"; do
+        IFS=':' read -r -a maps <<< "${i}"
+        if [[ ${maps[0]} == "${key}" ]]; then
+          info "Mapping ${key} into ${maps[1]}"
+          yq w -i "${TEMP_FILE}" -- "${maps[1]}" "$(yq r "${OLD_VALUES_YAML}" -- "${maps[0]}")"
+          yq d -i "${TEMP_FILE}" -- "${maps[0]}"
+        fi
+      done
+    else
+      yq w -i "${TEMP_FILE}" -- "${key}" "$(yq r "${OLD_VALUES_YAML}" -- "${key}")"
+    fi
+
+    for i in "${MAPPINGS_KEY_VALUE[@]}"; do
+      IFS=':' read -r -a maps <<< "${i}"
+      if [[ ${maps[0]} == "${key}" ]]; then
+        old_value=$(yq r "${OLD_VALUES_YAML}" -- "${key}")
+
+        if [[ ${maps[1]} == "${old_value}" ]]; then
+          info "Mapping ${key}'s value from ${maps[1]} into ${maps[2]} "
+          yq w -i "${TEMP_FILE}" -- "${maps[0]}" "${maps[2]}"
+        fi
+      fi
+    done
+
+  done
+  echo
+}
+
+function migrate_pre_upgrade_hook() {
+  # Keep pre-upgrade hook
+  if [[ -n "$(yq r "${TEMP_FILE}" -- sumologic.setup)" ]]; then
+    info "Updating setup hooks (sumologic.setup.*.annotations[helm.sh/hook]) to 'pre-install,pre-upgrade'"
+    yq w -i "${TEMP_FILE}" -- 'sumologic.setup.*.annotations[helm.sh/hook]' 'pre-install,pre-upgrade'
+  fi
+}
+
+function check_falco_state() {
```

Comment:
This one is not needed, we don't change the default falce state when updating`1.x -> 2.x`.